manifests/cache_dir.pp

```diff
@@ -54,7 +54,7 @@
     }
   }
 
-  if $facts['os']['selinux'] == true {
+  if has_key($facts['os'],'selinux') and $facts['os']['selinux']['enabled'] == true {
```

Comment:
I think this is equally effective.
```suggestion
  if fact('os.selinux.enabled') {
```

However, This should also always work
```suggestion
  if $facts['os']['selinux']['enabled'] {
```

(comment is also valid in other files, but I won't repeat the same comment multiple times)

Reply:
I just read over that you want to support Facter 2.4. In that case [fact()](https://forge.puppet.com/puppetlabs/stdlib/reference#fact) is the best thing to use.